pkg/placement/controller.go

```diff
@@ -441,7 +442,46 @@ func (c *Controller) reconcile(ctx context.Context, key util.Key) error {
 	}
 
 	c.logger.Info("Delivering", "object", util.GenerateObjectInfoString(obj), "to clusters", clusters)
-	deliverObjectToManagedClusters(c.logger, c.ocmClient, obj, clusters, managedByPlacements, c.wdsName, singletonStatus)
+	for _, clName := range clusters {
+		// find which placement(s) select this managedCluster
+		placementNames := []string{}
+		for _, plName := range managedByPlacements {
+			plObj, err := c.getPlacementByName(plName)
+			if err != nil {
```

Comment:
should we return an error if the error was for specific name? Maybe continue ?

Reply:
I think we should return. Thoughts are: If we continue, we may end up with an incomplete list of `placementNames` for current cluster, thus an inaccurate set of labels on the ManifestWork. But I'm not 100% sure. Maybe @pdettori has a better view on this?

Reply:
@waltforme overall looks fine, but could all the added logic be implemented in `deliverObjectToManagedClusters` (changing / rewriting that function and its signature as you see fit) ?

Reply:
Thanks for the suggestion. I packed the logic into a method (not a standalone function) on `c *Controller` because I see the logic relies on `c` in multiple places.